parsec/api/version.py

```diff
@@ -1,14 +1,62 @@
 # Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2016-2021 Scille SAS
 
-from collections import namedtuple
+from __future__ import annotations
+from typing import NamedTuple, Sequence, Tuple
 
 
-class ApiVersion(namedtuple("ApiVersion", "version revision")):
+class ApiVersion(NamedTuple):
+    version: int
+    revision: int
+
     def __str__(self) -> str:
         return f"{self.version}.{self.revision}"
 
+    @classmethod
+    def from_str(cls, value: str) -> ApiVersion:
+        raw_version, raw_revision = value.split(".")
+        version, revision = int(raw_version), int(raw_revision)
+        return cls(version, revision)
+
 
 API_V1_VERSION = ApiVersion(version=1, revision=3)
 API_V2_VERSION = ApiVersion(version=2, revision=7)
 API_V3_VERSION = ApiVersion(version=3, revision=1)
 API_VERSION = API_V3_VERSION
+
+
+class IncompatibleAPIVersionsError(Exception):
```

Comment:
Should inherit from `ProtocolError`

Reply:
That would create a `cyclic` dependencies, maybe I should move the exception to `api/protocol/base.py` ?

Reply:
done